bjoern/request.c

```diff
@@ -217,20 +225,25 @@ on_message_complete(http_parser* parser)
     _set_header(_REQUEST_METHOD, _GET);
   } else {
     _set_header_free_value(_REQUEST_METHOD,
-      PyString_FromString(http_method_str(parser->method)));
+      _Unicode_FromString(http_method_str(parser->method))
+      );
   }
 
   /* REMOTE_ADDR */
   _set_header(_REMOTE_ADDR, REQUEST->client_addr);
 
   PyObject* body = PyDict_GetItem(REQUEST->headers, _wsgi_input);
   if(body) {
-    /* We abused the `pos` member for tracking the amount of data copied from
-     * the buffer in on_body, so reset it to zero here. */
-    ((Iobject*)body)->pos = 0;
+    /* first do a seek(0) and then read() returns all data */
+    PyObject *buf = PyObject_CallMethodObjArgs(body, STR_seek, VALUE_zero,
+		    NULL);
+    Py_DECREF(buf); /* Discard the return value */
+    buf = PyObject_CallMethodObjArgs(body, STR_read, NULL);
+    Py_DECREF(body); /* Discard the BytesIO body object */
+    _set_header_free_value(_wsgi_input, buf); /* return the bytes to wsgi app */
```

Comment:
`wsgi.input` should be a file object, not a bytestring!

Reply:
Do we need to seek it to 0 before we hand over the bytesIO object?

Reply:
Yep!